test/luatest_helpers/vtest.lua

```diff
@@ -601,6 +601,117 @@ local function drop_instance(g, instance)
+        if tab[member] == nil then
+            error(string.format('timed out: %s is still nil', member))
+        end
+    end)
+end
+
+--------------------------------------------------------------------------------
+-- Service info helpers
+--------------------------------------------------------------------------------
+
+--
+-- The helper function which waits for the status of the service to be equal
+-- to the one user expects. Assumes, that the current status is not the one
+-- user want to see (requested status must have another status_idx).
+--
+-- Accepts timeout and function, which will be executed after checking of
+-- the status: opt.timeout and opts.callback
+--
+local function service_wait_for_new_status(service, status, opts)
+    local prev_status_idx = service.status_idx
+    t.helpers.retrying({timeout = opts.timeout or wait_timeout,
+                        delay = busy_step}, function()
+        if prev_status_idx ~= service.status_idx then
+            prev_status_idx = service.status_idx
+            if service.status == status then
+                return
+            end
+        end
+        if type(opts.callback) == 'function' then
+            opts.callback()
+        end
+        error(string.format('waiting for status "%s" timed out: ' ..
+                            'last status is "%s" with status_idx %d',
+                            status, service.status, service.status_idx))
+    end)
+end
+
+--
+-- Waits until the error, passed to the function's argument, occurs.
+-- If opts.is_instant then the function assumes that the error could
+-- already happen and check this situation. Oberwise, the function wait
+-- for a new status, different from the current one, to appear.
+--
+local function service_wait_for_error(service, error, opts)
+    local status_error = 'error'
+    opts = opts or {}
+    local compare
+    if type(service.error) == 'string' then
+        compare = function(t1, t2)
+            return pcall(t.assert_str_contains, t1, t2)
+        end
+    else
+        compare = function(t1, t2)
+            return t1 and t2 and type(t1) == type(t2) and
+                t1.code == t2.code and t1.message == t2.message
+        end
+    end
+    if opts.is_instant and service.status == status_error and
+         compare(service.error, error) then
+        return
+    end
+    while not compare(service.error, error) do
+        service_wait_for_new_status(service, status_error, opts)
+    end
+end
+
+--
+-- Waits until the status of the service is 'ok'. Ignores this status
+-- if `status_idx` equals to 0 as this one is default value and no iteration
+-- of background service was done yet. If not opts.is_instant then the
+-- function waits until the new ok status appears.
+--
+local function service_wait_for_ok(service, opts)
+    opts = opts or {}
+    local status_ok = 'ok'
+    if opts.is_instant and service.status_idx ~= 0 and
+                            service.status == status_ok then
```

Comment:
The alignment works so that next line's condition starts from the same place as the previous line condition. Which makes it -1 space from code, like in my diff above.